sdp/processors/modify_manifest/data_to_data.py

```diff
@@ -14,13 +14,232 @@
+        key_field: str = None,
+        target_samplerate: int = 16000,
+        target_nchannels: int = 1,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.input_field = input_field
+        self.output_field = output_field
+        self.key_field = key_field
+        self.resampled_audio_dir = resampled_audio_dir
+        self.target_samplerate = target_samplerate
+        self.target_nchannels = target_nchannels
+
+    def prepare(self):
+        os.makedirs(os.path.split(self.output_manifest_file)[0], exist_ok=True)
+        os.makedirs(self.resampled_audio_dir, exist_ok=True)
+
+    def process_dataset_entry(self, data_entry):
+        video = data_entry[self.input_field]
+        if self.key_field:
+            key = data_entry[self.key_field]
+            os.makedirs(os.path.join(self.resampled_audio_dir, key.split("/")[0]), exist_ok=True)
+        else:
+            key = os.path.splitext(video)[0].split("/")[-1]
+        audio = os.path.join(self.resampled_audio_dir, key) + ".wav"
+
+        if not os.path.isfile(audio):
+            ffmpeg_convert(video, audio, self.target_samplerate, self.target_nchannels)
+
+        data_entry[self.output_field]= audio
+        if self.key_field:
+            data_entry[self.key_field] = key
+        return [DataEntry(data=data_entry)]
+
+
+class ReadTxtLines(BaseParallelProcessor):
+    """
+    A class for reading text lines from a file and updating the dataset.
+
+    Args:
+    - input_field (str): The field containing the file path in the dataset.
+    - output_field (str): The field to store the read text lines in the dataset.
+    - **kwargs: Additional keyword arguments to be passed to the base class `BaseParallelProcessor`.
+
+    Methods:
+    - process_dataset_entry(data_entry): Processes a single dataset entry, reads text lines from the specified file, and updates the dataset.
+
+    Note:
+    - This class inherits from the `BaseParallelProcessor` class and extends its functionality to read text lines from a file and update the dataset.
+    """
+    def __init__(
+        self,
+        input_field: str,
+        output_field: str,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.input_field = input_field
+        self.output_field = output_field
+
+    def process_dataset_entry(self, data_entry):
+        fname = data_entry[self.input_field]
+        data_list = []
+        with open(fname, "r") as f:
+            for line in f:
+                line = line.strip()
+                if line:
+                    data = data_entry.copy()
+                    data[self.output_field] = line
+                    data_list.append(DataEntry(data=data))
+        return data_list
+
+
+class SplitLineBySentence(BaseParallelProcessor):
+    """
+    A class for splitting lines of text into sentences based on a specified pattern.
+
+    Args:
+    - input_field (str): The field containing the input text lines in the dataset.
```

Comment:
This class (and others) seems to be breaking away from the convention used so far in SDP. I would say that the current style of SDP processors would not use `input_field` and `output_field` but `text_key`, which would get overwritten. If you want to save the old information, you use the processor `DuplicateFields`. cc @Kipok in case you have opinions about the 2 different styles.

Reply:
changed